### PR TITLE
[SYCL][ESIMD] refactor timing code

### PR DESCRIPTION
**Pull request by cmc-rep**
Signed-off-by: Gang Y Chen <gang.y.chen@intel.com>

### CODE REVIEW

**Comment by cmc-rep** [comment]
@DenisBakhvalov , please review

File: SYCL/ESIMD/esimd_test_utils.hpp [hunk 1]
```diff
@@ -161,4 +162,30 @@ To bit_cast(const From &src) noexcept {
...
+    return std::chrono::duration_cast<Duration>(now - start_).count();
+  }
+
+private:
+  using Duration = std::chrono::duration<double>;
+  std::chrono::steady_clock::time_point start_;
+};
+
+// return msecond
+inline double report_time(const std::string &msg, event e0, event en) {
```
Comment (DenisBakhvalov) [inline SYCL/ESIMD/esimd_test_utils.hpp:183]: Maybe change the name to more descriptive like `get_elapsed_time`? 
I think it also makes sense to change the order of arguments: `event e0, event e1, const std::string &msg`.
Do we need two different event arguments? I see we pass the same event object every time.
And please add a description for the function.

**Review by DenisBakhvalov — COMMENTED** [review]
I left some comments, but overall it LGTM.

File: SYCL/ESIMD/esimd_test_utils.hpp (hunk 1 above)
Comment (cmc-rep) [reply SYCL/ESIMD/esimd_test_utils.hpp:183]: I think the name is fine. I have added some comment for report_time. two-event usage is in BitonicSortK and BitonicSortKv2

Comment (DenisBakhvalov) [reply SYCL/ESIMD/esimd_test_utils.hpp:183]: The `report_time` function does not report anything, so to me, the name looks confusing. Anyway, it should not block the PR.

**Review by DenisBakhvalov — APPROVED** [review]

**Comment by cmc-rep** [comment]
@vladimirlaz @DenisBakhvalov Could we merge this?  I think the jenkins is not stable. Last time, I retried, it worked, This time, it failed again, and I no longer can access the Jenkins

**Pull request merged**